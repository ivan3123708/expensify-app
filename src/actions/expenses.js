import uuid from 'uuid';
import db from '../firebase/firebase';

const addExpense = (expense = {}) => ({
  type: 'ADD_EXPENSE',
  expense: expense
});

const startAddExpense = (expenseData = {}) => {
  return (dispatch) => {

    const {
      description = '',
      note = '',
      amount = 0,
      createdAt = 0
    } = expenseData;

    const expense = {
      description: description,
      note: note,
      amount: amount,
      createdAt: createdAt
    };

    return db.ref('expenses').push(expense)
      .then((ref) => {
        dispatch(addExpense({
          id: ref.key,
          ...expense
        }));
      });
  }
}

const removeExpense = (id) => ({
  type: 'REMOVE_EXPENSE',
  id: id
});

const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id: id,
  updates: updates
});

export { addExpense, removeExpense, editExpense, startAddExpense };