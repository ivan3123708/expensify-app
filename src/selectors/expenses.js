import moment from 'moment';

const getFilteredExpenses = (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses.filter((expense) => {
    const startDateMatch = startDate ? startDate.isSameOrBefore(moment(expense.createdAt), 'day') : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(moment(expense.createdAt), 'day') : true;
    const textMatch = typeof text !== 'string' || expense.description.toLowerCase().includes(text.toLowerCase());

    return startDateMatch && endDateMatch && textMatch;
  }).sort((a, b) => {
    if (sortBy === 'date') return a.createdAt < b.createdAt;
    if (sortBy === 'amount') return a.amount < b.amount;
  });
}

export default getFilteredExpenses;