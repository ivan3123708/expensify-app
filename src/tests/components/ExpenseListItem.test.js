import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('Render expense list item', () => {
  const wrapper = shallow(<ExpenseListItem key={expenses[0].id} {...expenses[0]}/>);
  expect(wrapper).toMatchSnapshot();
});