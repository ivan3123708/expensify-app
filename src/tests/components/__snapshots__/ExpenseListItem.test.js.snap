// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render expense list item 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Cocaine
    </h3>
  </Link>
  <p>
    $500.00
    -
    January 1st, 1970
  </p>
</div>
`;
